File: 上學期/w9/Counting.c
#include <stdio.h>
#include <stdlib.h>

int arr[100005];

int main()
{
    int n, q;
    scanf("%d %d", &n, &q);

    for(int i=0; i<n; i++) scanf("%d", &arr[i]);
    while(q--)
    {
        int key, flag = 0;
        scanf("%d", &key);

        int right = n, left = 0, index;
        while(right != left)
        {
            index = (right+left)/2;
            if(arr[index] == key)
            {
                flag = 1;
                break;
            }
            else if(key < arr[index]) right = index;
            else left = index+1;
        }
        if(flag) printf("Yes\n");
        else printf("No\n");
    }
    return 0;
}

File: 下學期/W1/Counting.py
n, q = map(int, input().split())

nums = list(map(int, input().split()))

def binary_search(arr, left, right, hkey):
    while left <= right:
        mid = left + (right - left) // 2
        if arr[mid] == hkey:
            return mid
        elif arr[mid] < hkey:
            left = mid + 1
        elif arr[mid] > hkey:
            right = mid - 1
    return -1

for i in range(q):
    k = int(input())
    if binary_search(nums, 0, n-1, k) != -1:
        print("Yes")
    else:
        print("No")
